pilot/pkg/autoregistration/controller.go

```diff
@@ -215,14 +248,11 @@ func (c *Controller) RegisterWorkload(proxy *model.Proxy, conTime time.Time) err
 	if entryName == "" {
 		return nil
 	}
-	proxy.WorkloadEntryName = entryName
-	proxy.WorkloadEntryAutoCreated = autoCreate
 
-	c.mutex.Lock()
-	c.adsConnections[makeProxyKey(proxy)]++
-	c.mutex.Unlock()
+	proxy.SetWorkloadEntry(entryName, autoCreate)
+	c.adsConnections.Connect(conn)
```

Comment:
It is a little confusing, does it run this for all the connections? I think we should do this only for WLG related connections